google/cloud/aiplatform/models.py

```diff
@@ -1326,15 +1374,474 @@ def explain(
+            self._undeploy(deployed_model_id=deployed_model, sync=sync)
+
+        return self
+
+    def delete(self, force: bool = False, sync: bool = True) -> None:
+        """Deletes this Vertex AI Endpoint resource. If force is set to True,
+        all models on this Endpoint will be undeployed prior to deletion.
+
+        Args:
+            force (bool):
+                Required. If force is set to True, all deployed models on this
+                Endpoint will be undeployed first. Default is False.
+            sync (bool):
+                Whether to execute this method synchronously. If False, this method
+                will be executed in concurrent Future and any downstream object will
+                be immediately returned and synced when the Future has completed.
+
+        Raises:
+            FailedPrecondition: If models are deployed on this Endpoint and force = False.
+        """
+        if force:
+            self.undeploy_all(sync=sync)
+
+        super().delete(sync=sync)
+
+
+class PrivateEndpoint(Endpoint):
+    """
+    Represents a Vertex AI private Endpoint resource.
+
+    Read more [about private endpoints in the documentation.](https://cloud.google.com/vertex-ai/docs/predictions/using-private-endpoints)
+    """
+
+    def __init__(
+        self,
+        endpoint_name: str,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Retrieves a private Endpoint resource.
+
+        Example usage:
+            my_private_endpoint = aiplatform.PrivateEndpoint(
+                endpoint_name="projects/123/locations/us-central1/endpoints/my_endpoint_id"
+            )
+
+            or (when project and location are initialized)
+
+            my_private_endpoint = aiplatform.PrivateEndpoint(
+                endpoint_name="my_endpoint_id"
+            )
+
+        Args:
+            endpoint_name (str):
+                Required. A fully-qualified endpoint resource name or endpoint ID.
+                Example: "projects/123/locations/us-central1/endpoints/my_endpoint_id" or
+                "my_endpoint_id" when project and location are initialized or passed.
+            project (str):
+                Optional. Project to retrieve endpoint from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to retrieve endpoint from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        """
+        super().__init__(
+            endpoint_name=endpoint_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
```

Comment:
There should be validation that this resource is a private endpoint because it's possible to construct this class with a non-private endpoint.